owncloudApp/src/main/java/com/owncloud/android/presentation/transfers/TransferListComposeFragment.kt

```diff
@@ -0,0 +1,532 @@
+import androidx.documentfile.provider.DocumentFile
+import androidx.lifecycle.asFlow
+import androidx.work.WorkInfo
+import com.google.android.material.snackbar.Snackbar
+import com.owncloud.android.domain.transfers.model.TransferResult
+import com.owncloud.android.extensions.showMessageInSnackbar
+import com.owncloud.android.extensions.statusToStringRes
+import com.owncloud.android.lib.common.OwnCloudAccount
+import com.owncloud.android.presentation.authentication.AccountUtils
+import com.owncloud.android.ui.activity.FileActivity
+import com.owncloud.android.utils.MimetypeIconUtil
+import com.owncloud.android.workers.DownloadFileWorker
+import timber.log.Timber
+
+class TransferListComposeFragment : Fragment() {
+
+    private val transfersViewModel by viewModel<TransfersViewModel>()
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        return ComposeView(requireContext()).apply {
+            setContent {
+                setViewCompositionStrategy(ViewCompositionStrategy.DisposeOnViewTreeLifecycleDestroyed)
+                Transfers()
+            }
+        }
+    }
+
+    @Composable
+    fun Transfers() {
+        var transfersState by remember { mutableStateOf(emptyList<Pair<OCTransfer, OCSpace?>>()) }
+        val workInfos by transfersViewModel.workInfosListLiveData.asFlow().collectAsState(initial = emptyList())
+
+        LaunchedEffect(transfersViewModel) {
+            transfersViewModel.transfersWithSpaceStateFlow.collectLatest { transfers ->
+                transfersState = transfers
+            }
+        }
+
+        if (transfersState.isEmpty()) {
+            EmptyList()
+        } else {
+            TransferList(transfersWithSpace = transfersState, workInfo = workInfos)
+        }
+    }
+
+    @Composable
+    fun TransferList(transfersWithSpace: List<Pair<OCTransfer, OCSpace?>>, workInfo: List<WorkInfo>) {
+        val transfersGroupedByStatus = transfersWithSpace.groupBy { it.first.status }
+        LazyColumn {
+            transfersGroupedByStatus.forEach { (status, transfers) ->
+                item {
+                    UploadGroup(status = status, count = transfers.size)
+                }
+                transfers.sortedByDescending { it.first.transferEndTimestamp ?: it.first.id }
+                    .forEach { (transfer, space) ->
+                        item {
+                            TransferItem(transfer = transfer, space = space, workInfo = workInfo)
+                        }
+                    }
+            }
+        }
+    }
+
+    @Composable
+    fun TransferItem(transfer: OCTransfer, space: OCSpace?, workInfo: List<WorkInfo>) {
+        val remoteFile = File(transfer.remotePath)
+        var fileName = remoteFile.name
+        if (fileName.isEmpty()) {
+            fileName = File.separator
+        }
+
+        var path = "path/to/file"
```

Comment:
If you need to initialize this variable, I would set an empty string as initial value
In any case, I think you could do something like this:
```suggestion
        val path = remoteFile.parent?.let {
        ...
```

Reply:
Perfect! Now it's changed